reactive/kubernetes_worker.py

```diff
@@ -850,22 +882,25 @@ def render_and_launch_ingress():
     except CalledProcessError as e:
         hookenv.log(e)
         hookenv.log('Failed to create default-http-backend. Will attempt again next update.')  # noqa
-        hookenv.close_port(80)
-        hookenv.close_port(443)
-        return
+        # Now we need to know that we've failed to setup ingress controllers
+        # since ports are opened via configuration. We should advise the user
+        # to try again or close the ports
+        if "80" in hookenv.config().get("ports-to-open") or \
+           "443" in hookenv.config().get("ports-to-open"):
+            hookenv.log("Ingress controller disabled but ports 80 or 443 are kept open on ports-to-open option")
```

Comment:
A big reason to turn off ingress is to roll your own, which would open these ports as well. We should be careful here in complaining about something they may want to do.